mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java

```diff
@@ -196,6 +197,106 @@ public void checkParamsAndMetrics() {
   }
 
   @Test
+  public void testBatchedLogging() {
+    // Create exp
+    String expName = createExperimentName();
+    long expId = client.createExperiment(expName);
+    logger.debug(">> TEST.0");
+
+    // Test logging just metrics
+    {
+      RunInfo runCreated = client.createRun(expId);
+      String runUuid = runCreated.getRunUuid();
+      logger.debug("runUuid=" + runUuid);
+
+      client.logBatch(runUuid,
+              Arrays.asList(createMetric("met1", 0.081D, 10), createMetric("metric2", 82.3D, 100)),
```

Comment:
One observation, the `MlflowClient.logBatch` API is a little tricky to use as is given that it accepts protobuf arguments (the logic for `createMetric` is a little complex, involving using [protobuf builder](https://developers.google.com/protocol-buffers/docs/reference/java-generated#builders) APIs etc. In Python we have entity classes that make this easier and that the client APIs use (vs passing around generated protobuf classes). However in Java it seems the current pattern in MlflowClient is to use the protobuf classes, so the current API is good in that it follows the existing pattern. I also think adding a fluent API may simplify things a lot in the future too, so we might not need to worry for now. @aarondav may have thoughts / opinions here as well.

Reply:
That's a great idea. We can think of adding something like...

```
void logBatch(@NotNull String runUuid,
              @Nullable Map<String, Double> metrics,  // timestamp defaulted to sys time
              @Nullable Map<String, String> params,
              @Nullable Map<String, String> tags) {
 // create PB objects
 // call other logBatch API
}
```

Reply:
Added PR #1082 for this addition PR. Will merge in after this has been merged.